File: hugin/example/10-legend/legend_demo.ml
open Hugin

let () =
  (* Create figure *)
  let fig = Figure.create ~width:800 ~height:600 () in

  (* Add subplot *)
  let ax = Figure.add_subplot ~nrows:1 ~ncols:1 ~index:1 fig in

  (* Create some data *)
  let x = Nx.linspace Nx.float32 ~endpoint:true 0.0 10.0 100 in

  (* Plot multiple functions with labels *)
  let _ =
    Plotting.plot ~color:Artist.Color.blue ~label:"sin(x)" ~x ~y:(Nx.sin x) ax
  in
  let _ =
    Plotting.plot ~color:Artist.Color.red ~label:"cos(x)" ~x ~y:(Nx.cos x) ax
  in
  let _ =
    Plotting.plot ~color:Artist.Color.green ~linestyle:Artist.Dashed
      ~label:"sin(x) + cos(x)" ~x
      ~y:(Nx.add (Nx.sin x) (Nx.cos x))
      ax
  in

  (* Add scatter plot *)
  let x_scatter = Nx.linspace Nx.float32 ~endpoint:true 0.0 10.0 20 in
  let y_scatter = Nx.mul_s (Nx.sin x_scatter) 0.8 in
  let _ =
    Plotting.scatter ~c:Artist.Color.orange ~marker:Artist.Circle ~s:30.0
      ~label:"data points" ~x:x_scatter ~y:y_scatter ax
  in

  (* Enable legend *)
  let _ = Axes.legend true ax in

  (* Set labels and title *)
  let _ = Axes.set_xlabel "x" ax in
  let _ = Axes.set_ylabel "y" ax in
  let _ = Axes.set_title "Legend Demo: Trigonometric Functions" ax in

  (* Enable grid *)
  let _ = Axes.grid true ax in

  (* Save figure *)
  Printf.printf "Saving legend demo to legend_demo.png\n%!";
  Hugin.savefig "legend_demo.png" fig;
  Printf.printf "Done!\n%!"

File: hugin/examples/10-legend/legend_demo.py
import numpy as np
import matplotlib.pyplot as plt

# Create figure
fig, ax = plt.subplots(figsize=(8, 6))

# Create some data
x = np.linspace(0.0, 10.0, 100)

# Plot multiple functions with labels
ax.plot(x, np.sin(x), color='blue', label='sin(x)')
ax.plot(x, np.cos(x), color='red', label='cos(x)')
ax.plot(x, np.sin(x) + np.cos(x), color='green', linestyle='--', label='sin(x) + cos(x)')

# Add scatter plot
x_scatter = np.linspace(0.0, 10.0, 20)
y_scatter = np.sin(x_scatter) * 0.8
ax.scatter(x_scatter, y_scatter, color='orange', marker='o', s=30, label='data points')

# Enable legend
ax.legend()

# Set labels and title
ax.set_xlabel('x')
ax.set_ylabel('y')
ax.set_title('Legend Demo: Trigonometric Functions')

# Enable grid
ax.grid(True)

# Save figure
print("Saving legend demo to legend_demo_python.png")
plt.savefig('legend_demo_python.png', dpi=100)
print("Done!")